### DISCUSSION TITLE
Spike hub: Time measurement and angle tracking

### DISCUSSION THREAD

**Discussion by nightactive-git** [pybricks/support#1042] [General]
Hey everybody,
I really like this project and the idea of bringing Micropython to the Lego hubs. I started recently to play around with Pybricks on the Spike hub and realised, that most of the time measurement features from Micropython aren’t available on the Spike hub. The provided StopWatch feature ‘only’ provides milliseconds, which is not accurate enough for my use case. I want to track the yaw angle in order to drive in a straight line.

Is this a hardware limitation of the Spike hub?
Are there any other time measurement features available for Pybricks on the Spike hub?

I also saw that there is a method in pybricks for tracking the angle, which is not yet implemented. Does anybody know if there are technical reasons, why this feature is not yet implemented?

Kind regards!

**Comment by laurensvalk** [comment]
> I want to track the yaw angle in order to drive in a straight line.

> I also saw that there is a method in pybricks for tracking the angle, which is not yet implemented. Does anybody know if there are technical reasons, why this feature is not yet implemented?

You're in luck. We will release an update with the heading feature this coming week! :smile:

Would you still need the increased timer resolution in that case?

**Reply by nightactive-git** [reply]
Wow, really good news :) I'm looking forward to do some experiments with this new feature.

I think an increased timer resolution is always a good thing and can be used for many things. Is this possible? If yes: please go for it :)
